lib/shift_generator.rb

```diff
@@ -2,18 +2,19 @@
 require_relative 'offset_gen'
 
 class ShiftGenerator
-  attr_reader :key, :offset
+  attr_reader :key_gen, :offset_gen, :shift
 
   def initialize
-    @key = KeyGen.new
-    @offset = OffsetGen.new
+    @key_gen = KeyGen.new
+    @offset_gen = OffsetGen.new
   end
 
-  def shift_key(five_digit_key, last_four_digits_param)
-    random_key_hash = @key.create_key(five_digit_key)
-    date_offset_hash = @offset.create_offset(last_four_digits_param)
-    random_key_hash.merge(date_offset_hash) do |key, random_value, date_value|
-      random_value + date_value
+  def shift_key(five_digit_key, date)
+    key_param = @key_gen.create_key(five_digit_key)
+    date_param = @offset_gen.last_four_digits(@offset_gen.square_date(date))
+
+    key_param.zip(@offset_gen.create_offset(date_param)).map do |pair|
+      pair.sum
```

Comment:
this output was originally a hash but changed to an array to implement functionality